src/core/foundation.hpp

```diff
@@ -1339,12 +1343,23 @@ struct VariableResolver {
 
 template <typename T> T *varAsObjectChecked(const SHVar &var, const shards::Type &type) {
   SHTypeInfo typeInfo(type);
-  if (var.valueType != SHType::Object)
+  if (var.valueType != SHType::Object) {
+    auto err = fmt::format("Invalid type, expected: {} got: {}", type, magic_enum::enum_name(var.valueType));
+    assert(false);
```

Comment:
Any reason for this? Debuggers can also catch exception when they are thrown.

Reply:
I wanted to catch it every time, since it shouldn't happen if the program is written correctly.

Reply:
`SHLOG_FATAL` will crash the program via abort, should use that